tests/scripts/thread-cert/border_router/internet/test_upstream_dns.py

```diff
@@ -116,25 +110,57 @@ def test(self):
         router.srp_client_enable_auto_start_mode()
 
         # verify the server can forward the DNS query to upstream server.
-        self._verify_upstream_dns(br, router)
+        self._verify_upstream_dns(br, router, dns_server_a)
+        self._verify_recursive_dns(br, router, dns_server_b)
 
-    def _verify_upstream_dns(self, br, ed):
+    def _verify_upstream_dns(self, br, ed, dns_server):
         upstream_dns_enabled = br.dns_upstream_query_state
         if not upstream_dns_enabled:
             br.dns_upstream_query_state = True
         self.assertTrue(br.dns_upstream_query_state)
 
-        resolved_names = ed.dns_resolve(TEST_DOMAIN)
-        self.assertEqual(len(resolved_names), len(TEST_DOMAIN_IP6_ADDRESSES))
+        # Update BR's /etc/resolv.conf and force BR to reload it
+        dns_server_addr = dns_server.get_ether_addrs(ipv4=True, ipv6=False)[0]
+        br.bash(shlex.join(['echo', 'nameserver ' + dns_server_addr]) + ' > /etc/resolv.conf')
+
+        self.simulator.go(30)
+        resolved_names = ed.dns_resolve(TEST_DOMAIN_A)
+        self.assertEqual(len(resolved_names), len(TEST_DOMAIN_IP6_ADDRESSES_A))
         for record in resolved_names:
-            self.assertIn(ipaddress.IPv6Address(record[0]).compressed, TEST_DOMAIN_IP6_ADDRESSES)
+            self.assertIn(ipaddress.IPv6Address(record[0]).compressed, TEST_DOMAIN_IP6_ADDRESSES_A)
+
+    def _verify_recursive_dns(self, br, ed, dns_server):
+        recursive_dns_enabled = br.dns_upstream_query_state
+        if not recursive_dns_enabled:
+            br.dns_upstream_query_state = True
+        self.assertTrue(br.dns_upstream_query_state)
```

Comment:
Can we just do following after `srp_client_enable_auto_start_mode`?
```python
br.dns_upstream_query_state = True
self.assertTrue(br.dns_upstream_query_state)
```

It seems unnecessary to do this for both branches.